controller/admin_transactions_controller.php

```diff
@@ -296,11 +307,88 @@ private function do_action($action, $args)
 				}
 				// Clear $action status
 				$action = '';
+			break;
+			case 'add':
+				$errors = array();
+
+				$transaction_data = array(
+					'MT_ANONYMOUS'          => $this->request->is_set('u'),
+					'MT_USERNAME'           => $this->request->variable('username', '', true),
+					'MT_FIRST_NAME'         => $this->request->variable('first_name', '', true),
+					'MT_LAST_NAME'          => $this->request->variable('last_name', '', true),
+					'MT_PAYER_EMAIL'        => $this->request->variable('payer_email', '', true),
+					'MT_RESIDENCE_COUNTRY'  => $this->request->variable('residence_country', ''),
+					'MT_MC_GROSS'           => $this->request->variable('mc_gross', (float) 0),
+					'MT_MC_CURRENCY'        => $this->request->variable('mc_currency', ''),
+					'MT_MC_FEE'             => $this->request->variable('mc_fee', (float) 0),
+					'MT_PAYMENT_DATE_YEAR'  => $this->request->variable('payment_date_year', (int) $this->user->format_date(time(), 'Y')),
+					'MT_PAYMENT_DATE_MONTH' => $this->request->variable('payment_date_month', (int) $this->user->format_date(time(), 'n')),
+					'MT_PAYMENT_DATE_DAY'   => $this->request->variable('payment_date_day', (int) $this->user->format_date(time(), 'j')),
+					'MT_PAYMENT_TIME'       => $this->request->variable('payment_time', $this->user->format_date(time(), 'H:i:s')),
+					'MT_MEMO'               => $this->request->variable('memo', '', true),
+				);
+
+				if ($this->request->is_set_post('submit'))
+				{
+					try
+					{
+						$data_ary = $this->build_data_ary($transaction_data);
+						$this->ppde_actions->log_to_db($data_ary);
+
+						$this->log->add('admin', $this->user->data['user_id'], $this->user->ip, 'LOG_PPDE_MT_ADDED', array($transaction_data['MT_USERNAME']));
```

Comment:
Just after `'LOG_PPDE_MT_ADDED'` the `$log_time` argument is missing

Reply:
The time argument is optional. If omitted, the current `time()` is used.

Reply:
I know.
But since you added `array($transaction_data['MT_USERNAME'])`, this one is considered as an argument for `$log_time` and not for `$additionnal_data`. 😉

Reply:
You are absolutely right, thanks for pointing it out. :+1: